lib/stencil-push.utils.js

```diff
@@ -188,12 +190,54 @@ utils.deleteThemesIfNecessary = async (options) => {
     return options;
 };
 
+utils.checkIfDeletionIsComplete = () => {
+    return async.retryable(
+        {
+            interval: 1000,
+            errorFilter: (err) => {
+                if (err.message === 'ThemeStillExists') {
+                    console.log(`${'warning'.yellow} -- Theme still exists;Retrying ...`);
+                    return true;
+                }
+                return false;
+            },
+            times: 5,
+        },
+        utils.checkIfThemeIsDeleted(),
+    );
+};
+
+utils.checkIfThemeIsDeleted = () => async (options) => {
+    const {
+        themeLimitReached,
+        config: { accessToken },
+        storeHash,
+        themeIdsToDelete,
+    } = options;
+
+    if (!themeLimitReached) {
+        return options;
+    }
+
+    const apiHost = options.apiHost || options.config.apiHost;
+
+    const result = await themeApiClient.getThemes({ accessToken, apiHost, storeHash });
+
+    const themeStillExists = result.some((theme) => themeIdsToDelete.includes(theme.uuid));
+
+    if (themeStillExists) {
+        throw new Error('ThemeStillExists');
+    }
+
+    return { ...options, ...result };
```

Comment:
Do we need to include result here? Doesnt seem like it is being used

Reply:
oh yeah, that one is not needed! thanks

Reply:
Fixed :)